network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/NetpGeoswitchingActivity.kt

```diff
@@ -36,13 +43,12 @@ import kotlinx.coroutines.flow.onEach
 class NetpGeoswitchingActivity : DuckDuckGoActivity() {
     private val binding: ActivityNetpGeoswitchingBinding by viewBinding()
     private val viewModel: NetpGeoSwitchingViewModel by bindViewModel()
-    private lateinit var adapter: NetpGeoswitchingAdapter
+    private lateinit var lastSelectedButton: CompoundButton
```

Comment:
Had to manually handle deselection within the radio group because:
- The radiogroup items contains other views -> radio button, image, 2line list item, context menu
- for radio group to handle deselection properly, radio button should have unique ids.
- Since i am using viewbinding, this is not easy to do.

Reply:
@malmstein  this is the answer to question. It is was just more difficult to handle this in the recyclerview. There is possibly a way but in the interest of getting this fix out, moving away from recyclerview was easier.

Reply:
Makes sense, thanks for the context!